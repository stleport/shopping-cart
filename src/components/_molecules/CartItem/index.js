import React from 'react';
import PropTypes from 'prop-types';
import ProductPicture from '../../_atoms/ProductPicture';
import ProductDescription from '../ProductDescription';
import ProductButton from '../../_atoms/ProductButton';

const CartList = ({ 
  cartItem,
  addQuantity,
  subQuantity
}) => (
  (
    <div className="st-Item">
      <div className="st-Item__image--small">
        <ProductPicture 
          source={cartItem.images[0]} 
          altText={cartItem.title} 
        />
      </div>
      <div className="st-Item__content">
        <h2>{cartItem.title}</h2>
        <ProductDescription 
          description={cartItem.description} 
          quantity={cartItem.quantity} 
        />
        <div className="st-Item__extra">
          <ProductButton 
            iconName="minus"
            iconAlign="right" 
            iconColor="white"
            bgColor="dark"
            onClick={() => subQuantity(cartItem.productId, cartItem.quantity)}
          />
          <ProductButton 
            iconName="plus"
            iconAlign="right"
            iconColor="dark"
            bgColor="light"
            onClick={() => addQuantity(cartItem.productId)}
          />
        </div>
      </div>
    </div>
  )
);

CartList.propTypes = {
  cartItem: PropTypes.shape({
    // productId: PropTypes.number.isRequired,
    productId: PropTypes.number,
    quantity: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string)
  }).isRequired,
  addQuantity: PropTypes.func.isRequired,
  subQuantity: PropTypes.func.isRequired
};

export default CartList;
