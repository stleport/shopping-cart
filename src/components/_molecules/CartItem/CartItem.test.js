import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CartItem from '.';

function CartItemSetup(newProps = {}) {
  const props = {
    cartItem: { 
      title: 'Item',
      description: 'Description',
      images: ['http://example.com'],
      quantity: 1
    },
    addQuantity: jest.fn(),
    subQuantity: jest.fn(),
    removeFromCart: jest.fn()
  }; 

  const CartItemJSX = (
    <CartItem {...props} />
  );

  const CartItemComponent = shallow(CartItemJSX);
  return {
    CartItemComponent,
    CartItemJSX,
    props
  };
}

describe('CartItem', () => {
  it('renders correctly CartItem component', () => {
    const { CartItemJSX } = CartItemSetup();
    const CartItemComponent = renderer.create(CartItemJSX).toJSON();
    expect(CartItemComponent).toMatchSnapshot(); 
  });

  it('should contain two buttons', () => {
    const { CartItemComponent } = CartItemSetup();
    expect(CartItemComponent.find('ProductButton').length).toEqual(2);
  });
});
