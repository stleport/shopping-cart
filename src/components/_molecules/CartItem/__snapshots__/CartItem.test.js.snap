// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders correctly CartItem component 1`] = `
<div
  className="st-Item"
>
  <div
    className="st-Item__image--small"
  >
    <img
      alt="Item"
      className="image"
      onError={[Function]}
      src="http://example.com"
    />
  </div>
  <div
    className="st-Item__content"
  >
    <h2>
      Item
    </h2>
    <p
      className="description"
    >
      Description
    </p>
    <div
      className="st-Item__pricing"
    >
      Quantité : 
      1
    </div>
    <div
      className="st-Item__extra"
    >
      <button
        className="st-Edit-button__button st-Edit-button__button--dark buttontrue"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="st-Edit-button__icon--white"
          fill="white"
        >
          minus.svg
        </svg>
      </button>
      <button
        className="st-Edit-button__button st-Edit-button__button--light buttontrue"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="st-Edit-button__icon--dark"
          fill="dark"
        >
          plus.svg
        </svg>
      </button>
    </div>
  </div>
</div>
`;
